### ISSUE TITLE
Demo PC/JS doesn't work correctly in FireFox 

### ISSUE THREAD

**Issue by schors** [ionorg/ion#126]
Linux "Ubuntu 16.04.6 LTS"
75.0 (64-bit) Mozilla Firefox for Ubuntu canonical-1.0

I create room with more than one members
I expect watching all members streams

In Google Chrome all is Ok. But firefox shows me only LocalStream

There are some errors in developer's console:
Warning: This page uses the non standard property “zoom”. Consider using calc() in the relevant property values, or using “transform” along with “transform-origin: 0 0”
Warning: onaddstream is deprecated! Use peerConnection.ontrack instead.
Error: InvalidAccessError: Answer tried to set recv when offer did not set send

I'am not javascript ninja... I can't explain this behavior

**Comment by r-novel** [comment]
@schors Hi!
>Error: InvalidAccessError: Answer tried to set recv when offer did not set send

Several time ago, I've got similar error when I added 'recvonly' transceivers on js side and 'sendrecv' on pion server side; Firefox requires a match: recvonly <-> sendonly (sendonly <-> recvonly) or sendrecv <-> sendrecv.
Could you show me the offer/answer SDPs? 
In Firefox you can type 'about:webrtc' in the address bar for getting several information about your webrtc connections: sdps, ice candidates.

**Comment by schors** [comment]
SDP
Local SDP (Offer)
v=0
o=mozilla...THIS_IS_SDPARTA-75.0 8431642789634347329 0 IN IP4 0.0.0.0
s=-
t=0 0
a=sendrecv
a=fingerprint:sha-256 2D:94:95:9C:42:D1:E8:61:D5:54:44:F4:6F:65:48:AC:DA:B3:55:FF:01:BC:0D:6B:A0:8C:33:FE:C2:96:75:3C
a=group:BUNDLE 0 1
a=ice-options:trickle
a=msid-semantic:WMS *
m=audio 41918 UDP/TLS/RTP/SAVPF 109 9 0 8 101
c=IN IP4 192.168.0.105
a=candidate:0 1 UDP 2122252543 192.168.0.105 41918 typ host
a=candidate:2 1 UDP 2122187007 192.168.122.1 56700 typ host
a=candidate:4 1 UDP 2122055935 172.17.0.1 59679 typ host
a=candidate:6 1 UDP 2122121471 172.18.0.1 49244 typ host
a=candidate:8 1 TCP 2105524479 192.168.0.105 9 typ host tcptype active
a=candidate:9 1 TCP 2105458943 192.168.122.1 9 typ host tcptype active
a=candidate:10 1 TCP 2105327871 172.17.0.1 9 typ host tcptype active
a=candidate:11 1 TCP 2105393407 172.18.0.1 9 typ host tcptype active
a=candidate:0 2 UDP 2122252542 192.168.0.105 51820 typ host
a=candidate:2 2 UDP 2122187006 192.168.122.1 57149 typ host
a=candidate:4 2 UDP 2122055934 172.17.0.1 33941 typ host
a=candidate:6 2 UDP 2122121470 172.18.0.1 37947 typ host
a=candidate:8 2 TCP 2105524478 192.168.0.105 9 typ host tcptype active
a=candidate:9 2 TCP 2105458942 192.168.122.1 9 typ host tcptype active
a=candidate:10 2 TCP 2105327870 172.17.0.1 9 typ host tcptype active
a=candidate:11 2 TCP 2105393406 172.18.0.1 9 typ host tcptype active
a=recvonly
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:2/recvonly urn:ietf:params:rtp-hdrext:csrc-audio-level
a=extmap:3 urn:ietf:params:rtp-hdrext:sdes:mid
a=fmtp:109 maxplaybackrate=48000;stereo=1;useinbandfec=1
a=fmtp:101 0-15
a=ice-pwd:82326bcfc922847ca404546b42b0a4fb
a=ice-ufrag:eb888409
a=mid:0
a=rtcp:51820 IN IP4 192.168.0.105
a=rtcp-mux
a=rtpmap:109 opus/48000/2
a=rtpmap:9 G722/8000/1
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:101 telephone-event/8000
a=setup:actpass
a=ssrc:1290598139 cname:{d9741c70-6870-4521-9777-718e5070b5ef}
m=video 38749 UDP/TLS/RTP/SAVPF 120 121 126 97
c=IN IP4 192.168.0.105
a=candidate:0 1 UDP 2122252543 192.168.0.105 38749 typ host
a=candidate:2 1 UDP 2122187007 192.168.122.1 60241 typ host
a=candidate:4 1 UDP 2122055935 172.17.0.1 37102 typ host
a=candidate:6 1 UDP 2122121471 172.18.0.1 37099 typ host
a=candidate:8 1 TCP 2105524479 192.168.0.105 9 typ host tcptype active
a=candidate:9 1 TCP 2105458943 192.168.122.1 9 typ host tcptype active
a=candidate:10 1 TCP 2105327871 172.17.0.1 9 typ host tcptype active
a=candidate:11 1 TCP 2105393407 172.18.0.1 9 typ host tcptype active
a=candidate:0 2 UDP 2122252542 192.168.0.105 38953 typ host
a=candidate:2 2 UDP 2122187006 192.168.122.1 49766 typ host
a=candidate:4 2 UDP 2122055934 172.17.0.1 36813 typ host
a=candidate:6 2 UDP 2122121470 172.18.0.1 45791 typ host
a=candidate:8 2 TCP 2105524478 192.168.0.105 9 typ host tcptype active
a=candidate:9 2 TCP 2105458942 192.168.122.1 9 typ host tcptype active
a=candidate:10 2 TCP 2105327870 172.17.0.1 9 typ host tcptype active
a=candidate:11 2 TCP 2105393406 172.18.0.1 9 typ host tcptype active
a=recvonly
a=extmap:3 urn:ietf:params:rtp-hdrext:sdes:mid
a=extmap:4 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time
a=extmap:5 urn:ietf:params:rtp-hdrext:toffset
a=extmap:6/recvonly http://www.webrtc.org/experiments/rtp-hdrext/playout-delay
a=fmtp:126 profile-level-id=42e01f;level-asymmetry-allowed=1;packetization-mode=1
a=fmtp:97 profile-level-id=42e01f;level-asymmetry-allowed=1
a=fmtp:120 max-fs=12288;max-fr=60
a=fmtp:121 max-fs=12288;max-fr=60
a=ice-pwd:82326bcfc922847ca404546b42b0a4fb
a=ice-ufrag:eb888409
a=mid:1
a=rtcp:38953 IN IP4 192.168.0.105
a=rtcp-fb:120 nack
a=rtcp-fb:120 nack pli
a=rtcp-fb:120 ccm fir
a=rtcp-fb:120 goog-remb
a=rtcp-fb:121 nack
a=rtcp-fb:121 nack pli
a=rtcp-fb:121 ccm fir
a=rtcp-fb:121 goog-remb
a=rtcp-fb:126 nack
a=rtcp-fb:126 nack pli
a=rtcp-fb:126 ccm fir
a=rtcp-fb:126 goog-remb
a=rtcp-fb:97 nack
a=rtcp-fb:97 nack pli
a=rtcp-fb:97 ccm fir
a=rtcp-fb:97 goog-remb
a=rtcp-mux
a=rtpmap:120 VP8/90000
a=rtpmap:121 VP9/90000
a=rtpmap:126 H264/90000
a=rtpmap:97 H264/90000
a=setup:actpass
a=ssrc:4001410193 cname:{d9741c70-6870-4521-9777-718e5070b5ef}

**Comment by schors** [comment]
SDP
Local SDP (Offer)
v=0
o=mozilla...THIS_IS_SDPARTA-75.0 2331302848577713766 0 IN IP4 0.0.0.0
s=-
t=0 0
a=sendrecv
a=fingerprint:sha-256 4A:C3:ED:EA:D3:B5:3E:D2:5A:2B:F6:BB:B6:CE:2D:8D:D6:0B:53:76:A2:01:58:DC:D2:72:E3:D8:AF:57:3F:14
a=group:BUNDLE 0 1
a=ice-options:trickle
a=msid-semantic:WMS *
m=audio 48917 UDP/TLS/RTP/SAVPF 109 9 0 8 101
c=IN IP4 192.168.0.105
a=candidate:0 1 UDP 2122252543 192.168.0.105 48917 typ host
a=candidate:2 1 UDP 2122187007 192.168.122.1 40670 typ host
a=candidate:4 1 UDP 2122055935 172.17.0.1 52529 typ host
a=candidate:6 1 UDP 2122121471 172.18.0.1 43145 typ host
a=candidate:8 1 TCP 2105524479 192.168.0.105 9 typ host tcptype active
a=candidate:9 1 TCP 2105458943 192.168.122.1 9 typ host tcptype active
a=candidate:10 1 TCP 2105327871 172.17.0.1 9 typ host tcptype active
a=candidate:11 1 TCP 2105393407 172.18.0.1 9 typ host tcptype active
a=candidate:0 2 UDP 2122252542 192.168.0.105 37638 typ host
a=candidate:2 2 UDP 2122187006 192.168.122.1 58962 typ host
a=candidate:4 2 UDP 2122055934 172.17.0.1 49829 typ host
a=candidate:6 2 UDP 2122121470 172.18.0.1 59769 typ host
a=candidate:8 2 TCP 2105524478 192.168.0.105 9 typ host tcptype active
a=candidate:9 2 TCP 2105458942 192.168.122.1 9 typ host tcptype active
a=candidate:10 2 TCP 2105327870 172.17.0.1 9 typ host tcptype active
a=candidate:11 2 TCP 2105393406 172.18.0.1 9 typ host tcptype active
a=sendonly
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:2/recvonly urn:ietf:params:rtp-hdrext:csrc-audio-level
a=extmap:3 urn:ietf:params:rtp-hdrext:sdes:mid
a=fmtp:109 maxplaybackrate=48000;stereo=1;useinbandfec=1
a=fmtp:101 0-15
a=ice-pwd:2c2f562532fa26761b40277fd6ed4707
a=ice-ufrag:e9da819a
a=mid:0
a=msid:{f9d9ba2d-7c55-4f4e-864f-eff070c70a9e} {66f14cd1-9593-4646-96b7-d79ecfe5cd3a}
a=rtcp:37638 IN IP4 192.168.0.105
a=rtcp-mux
a=rtpmap:109 opus/48000/2
a=rtpmap:9 G722/8000/1
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:101 telephone-event/8000
a=setup:actpass
a=ssrc:3145462600 cname:{5e308ee4-dbee-4ffb-b06f-cf10ab219783}
m=video 46637 UDP/TLS/RTP/SAVPF 96
c=IN IP4 192.168.0.105
a=candidate:0 1 UDP 2122252543 192.168.0.105 46637 typ host
a=candidate:2 1 UDP 2122187007 192.168.122.1 33149 typ host
a=candidate:4 1 UDP 2122055935 172.17.0.1 36692 typ host
a=candidate:6 1 UDP 2122121471 172.18.0.1 53599 typ host
a=candidate:8 1 TCP 2105524479 192.168.0.105 9 typ host tcptype active
a=candidate:9 1 TCP 2105458943 192.168.122.1 9 typ host tcptype active
a=candidate:10 1 TCP 2105327871 172.17.0.1 9 typ host tcptype active
a=candidate:11 1 TCP 2105393407 172.18.0.1 9 typ host tcptype active
a=candidate:0 2 UDP 2122252542 192.168.0.105 34237 typ host
a=candidate:2 2 UDP 2122187006 192.168.122.1 34305 typ host
a=candidate:4 2 UDP 2122055934 172.17.0.1 60129 typ host
a=candidate:6 2 UDP 2122121470 172.18.0.1 44146 typ host
a=candidate:8 2 TCP 2105524478 192.168.0.105 9 typ host tcptype active
a=candidate:9 2 TCP 2105458942 192.168.122.1 9 typ host tcptype active
a=candidate:10 2 TCP 2105327870 172.17.0.1 9 typ host tcptype active
a=candidate:11 2 TCP 2105393406 172.18.0.1 9 typ host tcptype active
a=sendonly
a=extmap:3 urn:ietf:params:rtp-hdrext:sdes:mid
a=extmap:4 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time
a=extmap:5 urn:ietf:params:rtp-hdrext:toffset
a=extmap:6/recvonly http://www.webrtc.org/experiments/rtp-hdrext/playout-delay
a=ice-pwd:2c2f562532fa26761b40277fd6ed4707
a=ice-ufrag:e9da819a
a=mid:1
a=msid:{f9d9ba2d-7c55-4f4e-864f-eff070c70a9e} {79b5cdd2-216c-401d-9b39-be801cea9b6b}
a=rtcp:34237 IN IP4 192.168.0.105
a=rtcp-fb:96 ccm fir
a=rtcp-fb:96 nack
a=rtcp-fb:96 nack pli
a=rtcp-mux
a=rtpmap:96 VP8/90000
a=setup:actpass
a=ssrc:203289929 cname:{5e308ee4-dbee-4ffb-b06f-cf10ab219783}
Remote SDP (Answer)
v=0
o=- 850956249 1587216287 IN IP4 0.0.0.0
s=-
t=0 0
a=sendrecv
a=fingerprint:sha-256 AD:E1:00:AE:F9:4A:11:F9:13:8E:23:D1:B0:DB:86:43:90:8E:2E:06:B2:49:11:01:0B:5D:55:EA:BC:0D:6B:DF
a=group:BUNDLE 0 1
m=audio 9 UDP/TLS/RTP/SAVPF 111
c=IN IP4 0.0.0.0
a=candidate:foundation 1 udp 2130706431 172.18.0.7 5112 typ host generation 0
a=candidate:foundation 2 udp 2130706431 172.18.0.7 5112 typ host generation 0
a=candidate:foundation 1 udp 1694498815 94.237.10.234 5046 typ srflx raddr 0.0.0.0 rport 5046 generation 0
a=candidate:foundation 2 udp 1694498815 94.237.10.234 5046 typ srflx raddr 0.0.0.0 rport 5046 generation 0
a=recvonly
a=end-of-candidates
a=fmtp:111 maxplaybackrate=0;stereo=0;useinbandfec=1
a=ice-pwd:qUdDFQrIkwTonSyMoSqnujzwnYDRCknf
a=ice-ufrag:frxprkInyLKDUdLK
a=mid:0
a=rtcp-mux
a=rtpmap:111 opus/48000/2
a=setup:passive
m=video 9 UDP/TLS/RTP/SAVPF 96
c=IN IP4 0.0.0.0
a=candidate:foundation 1 udp 2130706431 172.18.0.7 5112 typ host generation 0
a=candidate:foundation 2 udp 2130706431 172.18.0.7 5112 typ host generation 0
a=candidate:foundation 1 udp 1694498815 94.237.10.234 5046 typ srflx raddr 0.0.0.0 rport 5046 generation 0
a=candidate:foundation 2 udp 1694498815 94.237.10.234 5046 typ srflx raddr 0.0.0.0 rport 5046 generation 0
a=recvonly
a=end-of-candidates
a=ice-pwd:qUdDFQrIkwTonSyMoSqnujzwnYDRCknf
a=ice-ufrag:frxprkInyLKDUdLK
a=mid:1
a=rtcp-fb:96 goog-remb
a=rtcp-fb:96 nack
a=rtcp-fb:96 nack pli
a=rtcp-mux
a=rtpmap:96 VP8/90000
a=setup:passive

**Comment by r-novel** [comment]
Hmm. SDPs from your second comment looks correctly. First of all, you could set Direction 'sendrecv' on Pion side: https://github.com/pion/ion/blob/master/pkg/rtc/transport/webrtctransport.go#L155
And js side: https://github.com/pion/ion/blob/master/sdk/js/src/Client.js#L275 . Rebuild example and try again.

I will try to reproduce this bug soon and give you feedback.

**Comment by schors** [comment]
I don't know why I see more tan one session strings on about:config. I don't understand what they mean. I try to use your suggestion and give feedback in half our

**Comment by schors** [comment]
> sendrecv

Hehe :)) No streams on both sides. But message about stream removing

**Comment by schors** [comment]
I checked Firefox 75.0 on Windows 10. It isn't works too.

**Comment by r-novel** [comment]
@schors 
>No streams on both sides. But message about stream removing

Could you check the RTP statistics in 'about:webrtc' ? If rtp packets have received, but stream not playback, I guess it might be problem with difference payload type.
For example: You sent from Chrome RTP packet with VP8 Payload Type (96), but Firefox waiting for another VP8 payload type number and stream will never playback.

**Comment by schors** [comment]
Two firefoxes doesn't work too. With and withaut patch. I try to investigate issue. There are no any streams with 'sendrecv' <-> 'sendrecv'. There is no Local Stream too

**Comment by schors** [comment]
Ops:
https://github.com/pion/webrtc/issues/717
But it is set of letters for me...

As a result ion demo doesn't work in FireFox...

**Comment by schors** [comment]
I went through all the combinations. It doesn't work...

**Comment by xobs** [comment]
I'm also unable to get it to work.  Trying on Firefox for Windows 10 and also Android.  I can see the "subscribe success" messages, and I can see the SFP exchanges, however there is an error on this line in `ion-conference.js`:

edit: I removed the line because `ion-conference.js` is just one big line.

**Comment by xobs** [comment]
It also does not work in Edge.  I get `InvalidAccessError` on `util.js(1,1)` as well as `SCRIPT5007: SCRIPT5007: Unable to get property 'stream' of undefined or null reference
ion-conference.js (27,16561)`

Edge also manages to do an SDP negotiation, however because of the other errors it does not work.

Browsers that I've tried that do not work:

* Firefox 75 Windows 10
* Edge 44.18362 on Windows 10
* Firefox 68.7.0 Android
* Chromium 75 Windows 10 (gives an Uncaught (in promise) DOMException error)

Browsers that I've tried that do work:

* Chrome 79 Android (I think? I only have one phone and this is the only one that seems to work, so I can't find anyone else to chat to)
